pkg/apis/application/v1alpha1/types.go

```diff
@@ -141,7 +141,12 @@ func (e Env) Envsubst(s string) string {
 		valByEnv[item.Name] = item.Value
 	}
 	return os.Expand(s, func(s string) string {
-		return valByEnv[s]
+		// allow escaping $ with $$
+		if s == "$" {
```

Comment:
This works because `$$` itself is expanded. We simply expand it to `$` so it's as if we escaped it.